### DISCUSSION TITLE
Error while uploading files to minio (s3)

### DISCUSSION THREAD

**Discussion by lluishi93** [directus/directus#5636] [Archive]
I'm trying to upload files to a minio (s3). This worked in v8, but un v9 we're getting the following error:

![image](https://user-images.githubusercontent.com/14786440/117963433-a85e2680-b320-11eb-9e13-b8714196b9eb.png)

To reproduce, just set up a clean directus install and try to upload a file, using the following envvars:

```
STORAGE_LOCATIONS: "minio"
STORAGE_MINIO_DRIVER: "s3"
STORAGE_MINIO_ROOT: "/pub/directus/originals"
STORAGE_MINIO_KEY: "22222"
STORAGE_MINIO_SECRET: "xxxxxxxxxxxxxxxxxxx"
STORAGE_MINIO_BUCKET: "22222"
STORAGE_MINIO_REGION: ""
STORAGE_MINIO_ENDPOINT: "https://minio.my.minio"
```

Version: RC64
Database: MySQL
Installation: Docker
Browser: Chrome 90

**Comment by rijkvanzanten** [comment]
I'm assuming aws-sdk is failing to read that undefined `REGION` in config 🤔 

The only other thing Minio requires is setting `s3ForcePathStyle` by adding:

```
STORAGE_MINIO_S3_FORCE_PATH_STYLE="true"
```

**Reply by lluishi93** [reply]
Should this be added to the docs maybe?

**Reply by lluishi93** [reply]
the point is that in Directus 8 we did not inform the region and it worked properly...

**Reply by rijkvanzanten** [reply]
> Should this be added to the docs maybe?

Looks like minio itself is already moving to support AWS's default connection method:

https://github.com/minio/minio/issues/7628
https://github.com/minio/minio/tree/master/docs/config#domain

> the point is that in Directus 8 we did not inform the region and it worked properly...

There's a difference between not setting the region, and setting it to an empty string though.

The following config seems to work as expected on my end:

```
STORAGE_MINIO_DRIVER="s3"
STORAGE_MINIO_KEY="minioadmin"
STORAGE_MINIO_SECRET="minioadmin"
STORAGE_MINIO_ENDPOINT="http://localhost:5106"
STORAGE_MINIO_BUCKET="example"
STORAGE_MINIO_S3_FORCE_PATH_STYLE="true"
```

**Reply by nachogarcia** [reply]
I think part of the issue here is that there is no easy way to debug these configuration problems.

In our case (with a config similar to yours rijk), after trying to upload a file, the whole directus crashes and needs to restart.

**Reply by lluishi93** [reply]
We tried not including the region to the envvars, but it stills fails to upload files :S Any idea? @rijkvanzanten

**Reply by nachogarcia** [reply]
STORAGE_MINIO_KEY: "22222"
STORAGE_MINIO_BUCKET: "22222"

In our case looks like directus was changing this to Integers, and needed to force it to be strings `STORAGE_MINIO_BUCKET: "string:22222"` https://docs.directus.io/reference/environment-variables/#environment-syntax-prefix

The region with value "" continues to work as expected.
https://docs.min.io/docs/how-to-use-aws-sdk-for-javascript-with-minio-server.html
